UnitySDK/Assets/ML-Agents/Examples/SharedAssets/Scripts/ProjectSettingsOverrides.cs

```diff
@@ -45,14 +45,14 @@ public void Awake()
         Physics.defaultSolverIterations = solverIterations;
         Physics.defaultSolverVelocityIterations = solverVelocityIterations;
 
-        var academy = FindObjectOfType<Academy>();
-        academy.LazyInitialization();
+        Academy.Instance.FloatProperties.RegisterCallback("gravity", f => { Physics.gravity = new Vector3(0, -f, 0); });
 
-        academy.FloatProperties.RegisterCallback("gravity", f => { Physics.gravity = new Vector3(0, -f, 0); });
+        Academy.Instance.ConnectToPlayerLoop();
```

Comment:
This handles stepping in most of the scenes.

Reply:
I would say the default should be to be connected and the user has to manually disconnect if they need to. I don't like having `Academy.Instance.ConnectToPlayerLoop();` being called in each Agent Initialize in some environments.
Also, since we do not have a use case for non-connected to player loop, maybe we should hide the functionality for now ?

Reply:
The problem is that the only interface for the Player Loop is to get the default loop, and set the current loop. There's no access to the current loop.
https://docs.unity3d.com/2019.2/Documentation/ScriptReference/Experimental.LowLevel.PlayerLoop.html

So ConnectToPlayerLoop will overwrite any changes that the use has already made to the player loop (or get overwritten, depending on which one gets called first). I think this is bad design on the PlayerLoop, but we have to work with it.

Reply:
I think it should remain as is.  Modification of the game loop is destructive to previous changes.